src/opm/output/eclipse/WriteRFT.cpp

```diff
@@ -160,6 +169,111 @@ namespace {
         }
     } // namespace RftUnits
 
+    std::optional<std::vector<Opm::data::Connection>::const_iterator>
+    findConnResults(const std::size_t                         cellIndex,
+                    const std::vector<Opm::data::Connection>& xcon)
+    {
+        auto pos = std::find_if(xcon.begin(), xcon.end(),
+                                [cellIndex](const Opm::data::Connection& xc)
+                                {
+                                    return xc.index == cellIndex;
+                                });
+
+        return (pos != xcon.end())
+            ? std::optional{pos} // C++17 class template argument deduction
+            : std::nullopt;
```

Comment:
>  Seems like argument deduction does not mix well with ternary operators

That's weird, because a selection of [online compilers (Godbolt.org)](https://godbolt.org/#z:OYLghAFBqd5TKALEBjA9gEwKYFFMCWALugE4A0BIEAZgQDbYB2AhgLbYgDkAjF%2BTXRMiAZVQtGIHgBYBQogFUAztgAKAD24AGfgCsp5eiyahUAUgBMAIUtXyKxqiIEh1ZpgDC6egFc2TA3cAGQImbAA5PwAjbFIQADZyAAd0JWIXJi9ffwMUtOchELDIthi4xIdsJwyRIhZSIiy/AJ57bEcCplr6oiKI6NiE%2BzqGppzWpRHe0P7SwfiASnt0H1JUTi5LAGZQ1F8cAGozLa8kzolj3DMtAEFrm9YOJSSWdaOAdhtbg5%2BDycwQCB0GcMhcTqEiJcDjQIBgmJMDlF0N4DuoFvdfh8vjdMZjSNgiKsmAcwGB1EctgAxP5EAFAkFCC6fHhmd4AEQOIBpdKYPno9GBkK22MxrLZ9zFEtuEIObBYoQgMvqwFQ5COAFYbJq5QBPGIAfR8vJUmDM6vF5oOqCQ9QAVAdlQA3M1atno26skW/OEIlg%2BEgHHUUjkw5WoCm4A5bdHCjG/fGE0jEnUAOhtSn1jokPmwEBj2LFXCW9G46v4AS4OnI6G4Hlstj%2BKzW2COFi2fHIRG0RaWAGsQOqAJwpraDgAcFh48THU/iWyHrRLXGk/DYIC20hTY60g4s7y2Wwng43Y8SFarNa4/CUIC0ne7SzgsBQcMEpDlTHWlGoGDYSQYsQ3rAv7/owpAAJIAnsxjAPqRCkEavbVgy8LcAAtP8xxsuY1i2Dw7xFuQsLoH%2BAEQQC9RsPE0jADwFgWHegoZDeXAYbSWE4TY1j4YRxGkWBkEgI6qBJEkmY8IO%2BoMfq6jUchnQsQA9P8aD1txBGPiBZGCcAg6tExQgsWxppbNhalWDxmkkaBsSCdBJiDlojEoUZmGmZxeEaUs2CECQpAEACi4wXZ5mGAQURAXmxaluW3bVrWQhvh%2BbwAGoENgADusQHBA%2BDEGQrbtgsBzqopPBlvelY9oY3AruQa7tlu6r7seFhjls%2B48BOsjnvwl7XrelU6I%2BiAQC%2B1lkd%2BfE2XElHHneOCOgQ6xpZlADySTMNwHZ0PQRCAdQURxVEoT1Dq238L%2BHDCGtTD0OdVXkDgH7AJIj2EPi1SOtgLFVtg6hVP6GwdhC7RxfQ4WkGdXg4HF8EEGuvCETQRjAEoq0ZRtW1I3IwhiBInAyLjigqBocX6LIRgmKpuHWGFEXwEsBmoaxblmbTFnvAcaFrVs17tFUnRuEwnjeM0gQi30JRlLkqTpEIYwtMkcudFLAxxBMAvVEI3SjGL4xtB0NRTGrcwa8MPSKwYkw9KbMs8EsShNusUjRVwFW9fFXAHDcABKACyBwquGw5bDlEO8miOV5X5hUO/wXZVQsbt1Q1WwjgemdZ1n5Ce/19iDYnw19iAFjxCm0jvOV6pjtIWhlxY0jlWONVcHzudxfnRe8aNo3jfxsRTVpYFoFTtF1wt2BLSt6WY5tFY7Qw%2B2kDeEBHY9J2sKQD0dldzBELd91xc9MFvVWH2CwQ32/fw/2A/tcWg0uVYQ1EUPbzDGxVvDiN8EsKMsDRhjLGC9%2BCCDxuISQRNwEkzUJoR6%2BggrUw8nTV%2BkUmYuXQmzFBnNua825mycIbJcBWAUAAcW5gAdQkPQKh/14IsG5ptTAxhnDhmMq/LCr8cH8yNq4CA7gratGCDMaWgxWh5HlpkfWStJGq1Eera2WtOi60aDIpRfCugmwUWba2UwhEWwaHbcRjtnaEzdh7Tu3BUSnjQtRK0Y8DgyBTFoFx0dfIFW2PHIa1UkDYBYDgOIeZyD9jrimPcPB8JDkHOqLQ7UtCU1qqudc6oUxzkHPEHg84MnzneGOd4Z4rFXgLnebuI14BjTQBNMCQ9qmDFovRJ6U9lrYGAfPC6Agl4HTXsdU628Ol7xundB659sAvTPvwC%2BX0fpxTvqgIGHSn7g0htDLAX8E7%2BV/sjVG6NZ4gI6TA/GUDZAwOUHA8m65DAwRplxOwaDGbyWYuhNaFgCFEJIeQ3hl9%2BGCPUcIyWOj7bK3yBkAxciMjGPNpUbWWjLZ/MNt82FRjAXiMMWo7ISsbbIuKIoh2yxVguzxUuSxj1Lw2PiHY6QQdUDhjouE9x%2BVSBxwWAnB8Sw/EBMGMEpcqd1yuPeDuaQp54jqniO8DJY51Tqg7qS7gA1SkPiIn3KpA8KBUGmmRISIkxKOgklJLQMk5KLRaW07Gi89rdPXlWTeZ0BkkWugfYZx8xmn3WU9Agn1nDX1mQDeZD8cZLMeq/d%2BOpP5w02Rdf%2BOzTWgOJkcwmJz5BnLJgg0uVzkGhXuZADBCluCKReQcZStIbmeS%2BTC4WosMUS0wJC2WIKFbwvBYUFFULlHGzhVWzWmjVG1omPo%2BFWLpg4t0Xip2BLzGtxJReaxskqVsCUMJA4urhwMQZbHLxLKfHJ3IBywJ1AU7JIPGk2JY5ByDneNIac0h1R7kKbK4pN4FVJxLvuFMN7a5NyHPud4Nctit3bnnOVW63YWGSTwJyMrp3FLKUsb6K8MggGkEAA%3D%3D%3D) seem to disagree with you.  It's critically dependent on compiling in C++17 mode, though.  That said, I don't know which compiler you're using so it might not have been covered by my testing.

> [using `if()`] should work nicely and might be more readable to some

I suppose.  I've pushed an update to that effect.

> One could also just return `std::optional<Opm::data::Connection>` as it will be dereferenced anyway

I would rather prefer not taking the copy.